src/engine/MaxConstraint.cpp

```diff
@@ -837,16 +665,35 @@ String MaxConstraint::serializeToString() const
 
     // Special delimiter ",e" represents elimination flag and variables
     output += Stringf( ",e" );
-    output += Stringf( ",%u", _eliminatedVariables ? 1 : 0 );
-    if ( _eliminatedVariables )
-        output += Stringf( ",%f", _maxValueOfEliminated );
+    output += Stringf( ",%u", _haveFeasibleEliminatedPhases ? 1 : 0 );
+    if ( _haveFeasibleEliminatedPhases )
+        output += Stringf( ",%f", _maxValueOfEliminatedPhases );
     else
         // Will be ignored in any case
         output += Stringf( ",%u", 0 );
 
     return output;
 }
 
+void MaxConstraint::eliminateCase( unsigned variable )
+{
+    if ( _cdInfeasibleCases )
+    {
+        markInfeasible( variableToPhase( variable ) );
+    }
+    else
+    {
+        _elements.erase( variable );
+        if ( _elementToAux.exists( variable ) )
+        {
+            unsigned aux = _elementToAux[variable];
+            _elementToAux.erase( variable );
+            _auxToElement.erase( aux );
+        }
+    }
+}
+
+
 bool MaxConstraint::haveOutOfBoundVariables() const
 {
     double fValue = _assignment.get( _f );
```

Comment:
Need to address aux variables here, too?

Reply:
Done.